ggshield/cmd/sca/scan.py

```diff
@@ -183,3 +200,28 @@ def get_sca_scan_all_filepaths(
     # Only sca_files field is useful in the case of a full_scan,
     # all the potential files already exist in `all_filepaths`
     return response.sca_files
+
+
+def sca_scan_diff(
```

Comment:
I think this function can be tested more:
- case of same refs
- empty repository
- repository without sca files (or maybe all SCA files were removed in ref)

Reply:
I added a test for the case of same refs. As discussed, I don't see any particular case to be tested for the other cases (that would be testing the backend).
Maybe these should be tested in integration tests though.